src/mui/components/dropdown/DropdownItem.tsx

```diff
@@ -4,36 +4,87 @@ import { green } from "@mui/material/colors";
 import ListItemIcon from "@mui/material/ListItemIcon";
 import ListItemText from "@mui/material/ListItemText";
 import MenuItem from "@mui/material/MenuItem";
-import React, { useCallback } from "react";
+import React, { useCallback, useState } from "react";
+import Menu from "@mui/material/Menu";
+
+export interface DropdownAction {
+    id: string;
+    disabled: boolean;
+    content: string | React.ReactElement;
+    icon: React.ReactElement;
+    shouldMenuStayOpened?: boolean;
+    key?: string;
+    showCheckIcon?: boolean;
+    isShown?: boolean;
+    isSelected?: boolean;
+    isDivider?: boolean;
+    subActions?: DropdownAction[];
+    onClick: (action: DropdownAction) => void;
+}
 
 export interface DropdownItemProps {
     disabled: boolean;
-    icon: React.ReactElement | null;
+    icon: React.ReactElement;
     id: string;
-    onClick: (id: string) => void;
+    onClick: (action: DropdownAction) => void;
     showCheckIcon?: boolean;
     content: string | React.ReactElement;
 }
 
 export function DropdownItem({
     disabled = false,
-    icon = null,
+    icon,
     id,
     onClick,
     showCheckIcon = false,
     content,
-}: DropdownItemProps) {
-    const onItemClick = useCallback(() => onClick(id), [id]);
+    subActions,
+}: DropdownItemProps & { subActions?: DropdownAction[] }) {
+    const [submenuOpen, setSubmenuOpen] = useState(false);
+    const handleMouseOver = useCallback(() => setSubmenuOpen(true), []);
+    const handleMouseOut = useCallback(() => setSubmenuOpen(false), []);
+    const handleClose = useCallback(() => setSubmenuOpen(false), []);
```

Comment:
handleOpen and handleClose would be enough instead of handleMouseOver, handleMouseOut .

Reply:
Yes. This is an erroneous change, should not have been committed.